tests/testthat/test-Samples-methods.R

```diff
@@ -23,3 +25,1302 @@ test_that("names-Samples returns correct names of the parameters", {
+  )
+  expect_error(
+    fitted <- fitPEM(samples, model, data, quantiles = c(0.025, 0.6, 975)),
+    "Assertion on 'quantiles' failed: Must have length 2, but has length 3."
+  )
+  expect_error(
+    fitted <- fitPEM(samples, model, data, quantiles = c(0.025, 0.975), hazard = "NotLogical"),
+    "Assertion on 'hazard' failed: Must be of type 'logical', not 'character'."
+  )
+})
+
+test_that("fitPEM-Samples-DALogisticLogNormal-DataDA works correctly", {
+  data <- DataDA(
+    x = c(0.1, 0.5, 1.5, 3, 6, 10, 10, 10),
+    y = c(0, 0, 1, 1, 0, 0, 1, 0),
+    doseGrid = c(0.1, 0.5, 1.5, 3, 6, seq(from = 10, to = 80, by = 2)),
+    u = c(42, 30, 15, 5, 20, 25, 30, 60),
+    t0 = c(0, 15, 30, 40, 55, 70, 75, 85),
+    Tmax = 60,
+    ID = 1L:8L,
+    cohort = as.integer(c(1:5, 6, 6, 6))
+  )
+  npiece_ <- 10
+  lambda_prior <- function(k) {
+    npiece_ / (data@Tmax * (npiece_ - k + 0.5))
+  }
+  model <- DALogisticLogNormal(
+    mean = c(-0.85, 1),
+    cov = matrix(c(1, -0.5, -0.5, 1), nrow = 2),
+    ref_dose = 56,
+    npiece = npiece_,
+    l = as.numeric(t(apply(as.matrix(c(1:npiece_), 1, npiece_), 2, lambda_prior))),
+    c_par = 2
+  )
+  options <- McmcOptions(
+    burnin = 500,
+    step = 2,
+    samples = 5000,
+    rng_kind = "Mersenne-Twister",
+    rng_seed = 225013
+  )
+  samples <- mcmc(data, model, options)
+
+  actual <- fitPEM(samples, model, data)
+  expect_snapshot(actual)
+
+  actual1 <- fitPEM(samples, model, data, middle = median)
+  expect_snapshot(actual1)
+
+  actual2 <- fitPEM(samples, model, data, quantiles = c(0.2, 0.8))
+  expect_snapshot(actual2)
+
+  actual3 <- fitPEM(samples, model, data, hazard = TRUE)
+  expect_snapshot(actual3)
+})
+
+test_that("plot-Samples-DALogisticNormal fails gracefully with bad input", {
+  data <- DataDA(
+    x = c(0.1, 0.5, 1.5, 3, 6, 10, 10, 10),
+    y = c(0, 0, 1, 1, 0, 0, 1, 0),
+    doseGrid = c(0.1, 0.5, 1.5, 3, 6, seq(from = 10, to = 80, by = 2)),
+    u = c(42, 30, 15, 5, 20, 25, 30, 60),
+    t0 = c(0, 15, 30, 40, 55, 70, 75, 85),
+    Tmax = 60,
+    ID = 1L:8L,
+    cohort = 1L:8L
+  )
+  npiece_ <- 10
+  lambda_prior <- function(k) {
+    npiece_ / (data@Tmax * (npiece_ - k + 0.5))
+  }
+
+  model <- DALogisticLogNormal(
+    mean = c(-0.85, 1),
+    cov = matrix(c(1, -0.5, -0.5, 1), nrow = 2),
+    ref_dose = 56,
+    npiece = npiece_,
+    l = as.numeric(t(apply(as.matrix(c(1:npiece_), 1, npiece_), 2, lambda_prior))),
+    c_par = 2
+  )
+  options <- McmcOptions(
+    burnin = 100,
+    step = 2,
+    samples = 1000,
+    rng_kind = "Mersenne-Twister",
+    rng_seed = 552914
+  )
+  samples <- mcmc(data, model, options)
+
+  expect_error(
+    plot(samples, model, data, showLegend = "NotLogical"),
+    "Assertion on 'showLegend' failed: Must be of type 'logical', not 'character'."
+  )
+  expect_error(
+    plot(samples, model, data, hazard = "NotLogical"),
+    "Assertion on 'hazard' failed: Must be of type 'logical', not 'character'."
+  )
+})
+
```

Comment:
to ease navigation in RStudio, can you please add little sections a la

`# plot-Samples-DALogisticNormal ----`

etc. in this test file?

Other test files also have that, see e.g https://github.com/Roche/crmPack/blob/main/tests/testthat/test-Model-class.R